File: src/my_project/easy_problems/from151To200/remove_adjacent_duplicates_string.py
from typing import List 

class Solution:
    def removeDuplicates(self, s: str) -> str:

        list_solution = []

        for letter in s:

            if not list_solution:
                list_solution.append(letter)

            elif list_solution and list_solution[-1] != letter: 
                list_solution.append(letter)

            else: 
                list_solution.pop()


        return ''.join(list_solution)


solution = Solution()

print(solution.removeDuplicates(s = "abbaca"))

print(solution.removeDuplicates(s = "azxxzy"))

File: my_project_typescript/easy_problems_typescript/from51To100/removeAdjacentDuplicatesString.ts
function removeDuplicates(s: string): string {
    let listSolution: Array<string> = []

    for (let letter of s){
        if (listSolution.length ==0){
            listSolution.push(letter)
        }
        else if(listSolution.length > 0 && listSolution[listSolution.length-1] !== letter){
            listSolution.push(letter)
        
        }
        else{
            listSolution.pop()
        }
    }
    return listSolution.join('')
};

console.log(removeDuplicates("abbaca"))

console.log(removeDuplicates("azxxzy"))


